source/imaer-gml/src/main/java/nl/overheid/aerius/gml/GMLCalculationSetOptionsReader.java

```diff
@@ -68,8 +69,21 @@ public CalculationSetOptions readCalculationSetOptions(final Theme theme) {
       OptionsMetadataUtil.addOptionsFromMap(theme, optionsMap, options);
     }
 
-    options.setCalculationType(CalculationType.safeValueOf(calculationMetaData.getCalculationType()));
+    setCalculationMethod(calculationMetaData, options);
     options.setCalculateMaximumRange(calculationMetaData.getMaximumRange() == null ? 0.0 : calculationMetaData.getMaximumRange());
     return options;
   }
+
+  /**
+   * Set the calculation method. If calculation method is null try if calculation type was used.
+   *
+   * @param calculationMetaData
+   * @param options
+   */
+  private void setCalculationMethod(final IsCalculationMetaData calculationMetaData, final CalculationSetOptions options) {
+    final String method = calculationMetaData.getCalculationMethod();
+
+    options.setCalculationMethod(
+        method == null ? CalculationType.toCalculationMethod(method) : CalculationMethod.safeValueOf(calculationMetaData.getCalculationMethod()));
```

Comment:
```suggestion
        method == null ? CalculationType.toCalculationMethod(calculationMetaData.getCalculationType()) : CalculationMethod.safeValueOf(method));
```
Not entirely sure if this is correct, but as it is it doesn't seem right either.

Reply:
Should be as you suggest.